cluster-autoscaler/core/scale_down.go

```diff
@@ -266,6 +268,11 @@ func (sd *ScaleDown) TryToScaleDown(nodes []*apiv1.Node, pods []*apiv1.Pod, pdbs
 	// to recreate on other nodes.
 	emptyNodes := getEmptyNodes(candidates, pods, sd.context.MaxEmptyBulkDelete, sd.context.CloudProvider)
 	if len(emptyNodes) > 0 {
+		nodeDeletionStart := time.Now()
+		defer func(nodeDeletionStart time.Time, nodeDeletionDuration *time.Duration) {
+			stop := time.Now()
+			*nodeDeletionDuration = stop.Sub(nodeDeletionStart)
+		}(nodeDeletionStart, &nodeDeletionDuration)
```

Comment:
Good point. I think the best solution here would be to move deleting empty nodes to a separate function (similar to how we have deleteNode function). Than we get to keep the start and stop close to each other without resorting to hacks.

As a side benefit that would slightly reduce the pain I feel every time I read our central functions, like RunOnce, TryToScaleDown or ScaleUp and once again see how bloated and hard to follow they are.